### PR TITLE
[FIX] mzid: fall back to C/N-term modification because of lack of sup…

### PR DESCRIPTION
**Pull request by timosachsenberg**
…port

for protein C/N-term in OpenMS

addresses https://github.com/OpenMS/OpenMS/issues/2445

### CODE REVIEW

File: src/openms/source/FORMAT/HANDLERS/MzIdentMLDOMHandler.cpp [hunk 1]
```diff
@@ -957,12 +957,16 @@ namespace OpenMS
                       }
                       else if (spci->second.front().getAccession() == "MS:1002057")  // pro nterm
                       {
-                        ResidueModification m = ModificationsDB::getInstance()->getModification(mname,  r, ResidueModification::PROTEIN_N_TERM);
+                        // TODO: add support for protein N-terminal modifications in unimod
+                        // ResidueModification m = ModificationsDB::getInstance()->getModification(mname,  r, ResidueModification::PROTEIN_N_TERM);
+                        ResidueModification m = ModificationsDB::getInstance()->getModification(mname,  r, ResidueModification::N_TERM);
```
Comment (hendrikweisser) [inline src/openms/source/FORMAT/HANDLERS/MzIdentMLDOMHandler.cpp]: Double space after "mname,"

File: src/openms/source/FORMAT/HANDLERS/MzIdentMLDOMHandler.cpp [hunk 2]
```diff
@@ -957,12 +957,16 @@ namespace OpenMS
...
                         mod = m.getFullId();
                       }
                       else if (spci->second.front().getAccession() == "MS:1002058")  // pro cterm
                       {
-                        ResidueModification m = ModificationsDB::getInstance()->getModification(mname,  r, ResidueModification::PROTEIN_C_TERM);
+                        // TODO: add support for protein C-terminal modifications in unimod
+                        // ResidueModification m = ModificationsDB::getInstance()->getModification(mname,  r, ResidueModification::PROTEIN_C_TERM);
+                        ResidueModification m = ModificationsDB::getInstance()->getModification(mname,  r, ResidueModification::C_TERM);
```
Comment (hendrikweisser) [inline src/openms/source/FORMAT/HANDLERS/MzIdentMLDOMHandler.cpp]: Double space again.

File: src/openms/source/FORMAT/HANDLERS/MzIdentMLDOMHandler.cpp [hunk 3]
```diff
@@ -957,12 +957,16 @@ namespace OpenMS
                       }
                       else if (spci->second.front().getAccession() == "MS:1002057")  // pro nterm
                       {
-                        ResidueModification m = ModificationsDB::getInstance()->getModification(mname,  r, ResidueModification::PROTEIN_N_TERM);
+                        // TODO: add support for protein N-terminal modifications in unimod
```
Comment (hendrikweisser) [inline src/openms/source/FORMAT/HANDLERS/MzIdentMLDOMHandler.cpp]: "in unimod" makes it sound like Unimod lacks support for this. Better "from Unimod"?

File: src/openms/source/FORMAT/HANDLERS/MzIdentMLDOMHandler.cpp [hunk 4]
```diff
@@ -957,12 +957,16 @@ namespace OpenMS
...
+                        // ResidueModification m = ModificationsDB::getInstance()->getModification(mname,  r, ResidueModification::PROTEIN_N_TERM);
+                        ResidueModification m = ModificationsDB::getInstance()->getModification(mname,  r, ResidueModification::N_TERM);
                         mod = m.getFullId();
                       }
                       else if (spci->second.front().getAccession() == "MS:1002058")  // pro cterm
                       {
-                        ResidueModification m = ModificationsDB::getInstance()->getModification(mname,  r, ResidueModification::PROTEIN_C_TERM);
+                        // TODO: add support for protein C-terminal modifications in unimod
```
Comment (hendrikweisser) [inline src/openms/source/FORMAT/HANDLERS/MzIdentMLDOMHandler.cpp]: See above ("in" -> "from").

**Comment by hendrikweisser** [comment]
Thanks for fixing this, Timo.
Could you add a test case that would have caught the problem?

**Comment by timosachsenberg** [comment]
related to https://github.com/OpenMS/OpenMS/issues/1913

**Comment by mwalzer** [comment]
:+1:  for merge

**Pull request merged**